source/mixin-class.d.ts

```diff
@@ -0,0 +1,72 @@
+import type {Class, Constructor} from './basic';
+
+/**
+A type for mixin classes, particularly useful when applying [ECMAScript decorators](https://www.typescriptlang.org/docs/handbook/mixins.html#decorators-and-mixins-4881).
+
+@typeParam AddedStatic - static fields for the mixin.
+@typeParam AddedInstance - fields for instances of the mixin.
+@typeParam BaseClass - the type of your base class.
+@typeParam Arguments - arguments for your constructor.
+
+@category Class
+ */
+export type MixinClass<AddedStatic, AddedInstance, BaseClass extends Class<unknown>, Arguments extends unknown[] = ConstructorParameters<BaseClass>> = (
```

Comment:
I've been tinkering with types that derive from this, and I'm now thinking `AddedStatic` and `AddedInstance` should extend `object`.  I mean, why in the world would we want to have a mixin where the added static fields extend `boolean`?